src/main.rs

```diff
@@ -189,16 +189,16 @@ impl TestConfig {
 
 fn read_config(path: &str, symfile: &HashMap<String, (u32, u16)>) -> (TestConfig, Vec<TestConfig>) {
 	fn parse_u8(value: &toml::Value, hint: &str) -> Option<u8> {
-		if let toml::Value::Integer(value) = value {
-			if *value < 256 && *value >= -128 {
-				Some(*value as u8)
-			} else {
+		match value {
+			toml::Value::Integer(value) if -128 <= *value && *value < 256 => Some(*value as u8),
```

Comment:
I'm not so sure about these pattern guards. I guess they can improve readability, but they're also not super ergonomic...